server/protocols/jmap/src/main/java/org/apache/james/jmap/AuthenticationFilter.java

```diff
@@ -84,19 +82,17 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
 
     }
 
-    private Stream<String> getAuthHeaders(HttpServletRequest httpRequest) {
-        Enumeration<String> authHeaders = httpRequest.getHeaders(AUTHORIZATION_HEADERS);
-
-        return authHeaders != null ? Collections.list(authHeaders).stream() : Stream.of();
-    }
-
-    private HttpServletRequest addSessionToRequest(HttpServletRequest httpRequest, MailboxSession mailboxSession) {
-        httpRequest.setAttribute(MAILBOX_SESSION, mailboxSession);
+    private HttpServletRequest addSessionToRequest(HttpServletRequest httpRequest, Optional<MailboxSession> mailboxSession) {
+        httpRequest.setAttribute(MAILBOX_SESSION, mailboxSession.get());
         return httpRequest;
     }
 
-    private MailboxSession createSession(AuthenticationStrategy authenticationMethod, Stream<String> authorizationHeaders) {
-        return authenticationMethod.createMailboxSession(authorizationHeaders);
+    private Optional<MailboxSession> createSession(AuthenticationStrategy authenticationMethod, HttpServletRequest httpRequest) {
```

Comment:
It's a rare case where I would use Stream to be able to flatmap afterward without Optional Hell